### PR TITLE
Universe events stats handle universe identity proof type

### PR DESCRIPTION
**Pull request by ffranr**
Depends on https://github.com/lightninglabs/taproot-assets/pull/540 . Do not merge this PR before merging https://github.com/lightninglabs/taproot-assets/pull/540 I've converted this PR to draft until all parent PRs have been merged.

This PR adds universe identity proof type support to universe events stats database SQL statements.

### CODE REVIEW

File: tapdb/universe_stats.go [hunk 1]
```diff
@@ -141,6 +141,7 @@ func (u *UniverseStats) LogSyncEvent(ctx context.Context,
 			EventTimestamp: u.clock.Now().UTC().Unix(),
 			AssetID:        uniID.AssetID[:],
 			GroupKeyXOnly:  groupKeyXOnly,
+			ProofType:      uniID.ProofType.String(),
```
Comment (Roasbeef) [inline tapdb/universe_stats.go:144]: Would've expected these test run on top of that branch to trigger unit test failures within it?

File: tapdb/sqlc/queries/universe.sql [hunk 2]
```diff
@@ -113,14 +113,18 @@ SELECT * FROM universe_servers;
...
 WITH group_key_root_id AS (
     SELECT id
-    FROM universe_roots
+    FROM universe_roots roots
     WHERE group_key = @group_key_x_only
+      AND roots.proof_type = @proof_type
 ), asset_id_root_id AS (
     SELECT leaves.universe_root_id AS id
     FROM universe_leaves leaves
+    JOIN universe_roots roots
```
Comment (Roasbeef) [inline tapdb/sqlc/queries/universe.sql:122]: Ah actually a bug fix? Or I guess we'd still have the same return value, this join is just slightly more correct.

File: tapdb/sqlc/queries/universe.sql [hunk 3]
```diff
@@ -142,14 +142,18 @@ INSERT INTO universe_events (
...
+        AND roots.proof_type = @proof_type
 ), asset_id_root_id AS (
     SELECT leaves.universe_root_id AS id
     FROM universe_leaves leaves
-             JOIN genesis_info_view gen
-                  ON leaves.asset_genesis_id = gen.gen_asset_id
+    JOIN universe_roots roots
+        ON leaves.universe_root_id = roots.id
+    JOIN genesis_info_view gen
+        ON leaves.asset_genesis_id = gen.gen_asset_id
```
Comment (Roasbeef) [inline tapdb/sqlc/queries/universe.sql:154]: Out of place, but I think we need to decide how we want to display things on the RPC level here. If we're going with the split approach, do we also make further RPC changes or just try to unify things? Some input from @jamaljsr likely useful here.

File: tapdb/sqlc/migrations/000007_universe.up.sql [hunk 4]
```diff
@@ -80,8 +80,7 @@ CREATE VIEW universe_stats AS
         COUNT(CASE WHEN u.event_type = 'SYNC' THEN 1 ELSE NULL END) AS total_asset_syncs,
         COUNT(CASE WHEN u.event_type = 'NEW_PROOF' THEN 1 ELSE NULL END) AS total_asset_proofs,
         roots.asset_id,
-        roots.group_key,
-        roots.namespace_root
```
Comment (Roasbeef) [inline tapdb/sqlc/migrations/000007_universe.up.sql:84]: It wasn't used elsewhere?

**Review by Roasbeef — COMMENTED** [review]
SGTM for the split approach. 

I think we still need to decide if we go with split or unified on the RPC layer.

File: tapdb/sqlc/migrations/000007_universe.up.sql (hunk 4 above)
Comment (guggero) [reply tapdb/sqlc/migrations/000007_universe.up.sql:84]: I guess it was just in the list for completeness, but the actual change here is that it's removed from the `GROUP BY` clause.

**Review by guggero — COMMENTED** [review]
Looks good as far as I can tell. Will do some manual testing once the base PR has been merged.

File: tapdb/universe_stats.go (hunk 1 above)
Comment (ffranr) [reply tapdb/universe_stats.go:143]: Yes me too, but the stats must be aggregated correctly regardless, or we were missing coverage.

File: tapdb/sqlc/queries/universe.sql (hunk 2 above)
Comment (ffranr) [reply tapdb/sqlc/queries/universe.sql:122]: This additional `join` ensures that we select the asset ID from the row which has the correct proof type, but functionally it makes no difference.

File: tapdb/sqlc/queries/universe.sql (hunk 3 above)
Comment (ffranr) [reply tapdb/sqlc/queries/universe.sql:154]: @Roasbeef the changes in this PR don't lead to a split approach from @jamaljsr 's perspective. The query functions still aggregate across universe proof type.

**Comment by ffranr** [comment]
@Roasbeef stats are still aggregated as before. This PR adds proof type resolution to how we store stats, but stats are still queried and aggregated as before.

**Review by guggero — COMMENTED** [review]
Did a bunch of manual testing and found a bunch of small things that should be fixed.
Summarized them in this commit (feel free to just take over/split into fixup commits): https://github.com/lightninglabs/taproot-assets/tree/561-fixups

With those fixes applied, I think this is ready to go.

File: rpcserver.go [hunk 5]
```diff
@@ -3200,9 +3200,8 @@ func (r *rpcServer) QueryProof(ctx context.Context,
 		return nil, err
 	}
 
-	rpcsLog.Debugf("[QueryProof]: fetching proof at "+
-		"(universeID=%x, leafKey=%x)", universeID,
-		leafKey.UniverseKey())
+	rpcsLog.Debugf("[QueryProof]: fetching proof at (universeID=%x, "+
```
Comment (Roasbeef) [inline rpcserver.go:3203]: This should also be `%v`.

**Review by Roasbeef — APPROVED** [review]
LGTM

**Review by guggero — APPROVED** [review]

**Pull request merged**